src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php

```diff
@@ -132,7 +137,11 @@ protected function injectToolbar(Response $response, Request $request, array $no
             ))."\n";
             $content = substr($content, 0, $pos).$toolbar.substr($content, $pos);
             $response->setContent($content);
+
+            return;
         }
+
+        $this->logger->warning('No body tag was found. Please add one to display the toolbar.');
```

Comment:
`No closing body tag ...`

AFAIK we never use a `NullLogger`, in favor of `if (null !== $this->logger) { ... }`

Reply:
what about using `debug()` level instead :/

Reply:
I took the example in the Messenger component : https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Messenger/Middleware/HandleMessageMiddleware.php#L37

Which strategy would be better ? Inject the service ? Or the make the condition ?